core/templates/dev/head/pages/exploration_player/SupplementalCardDirective.js

```diff
@@ -72,14 +71,18 @@ oppia.directive('supplementalCard', [
 
           $scope.CONTINUE_BUTTON_FOCUS_LABEL = CONTINUE_BUTTON_FOCUS_LABEL;
 
-          $scope.helpCardHtml = null;
-          $scope.helpCardHasContinueButton = false;
+          $scope.helpCardHtml = HelpCardService.getHtmlHelpCard().helpCardHtml;
```

Comment:
Could you find a way to bind these to properties of the service? I feel like you can cut down a lot of the code that way by not having to simultaneously update the $scope values as well as clearing/setting the service.

Reply:
As per my understanding, by "bind" you mean two-way binding between $scope and HelpCardService. I completely agree that two-way binding will reduce a lot of codes. 
But the problem is I can't find a way through which the changes in the $scope object will be reflected in the service synchronously (to alter clearing and setting). Although vice-versa can be done, and I think can eliminate getHelpCard function. I need some input on this...
What do you think @seanlip ?

Reply:
I don't know why these two statements are being changed at all, actually. They run when the directive's initialized, at which point there's no help card to show yet.

Reply:
So the problem is whenever an interaction with supplementalCard is loaded, the $scope object get reinitialized. And therefore, when we go to the previous card and come back to 4th Card $scope.helpCardHtml gets reinitialized to null, no matter what its value was before. This makes the supplement card to disappear even if it was visible before.
So, what I thought let's not initialize $scope.helpCardHtml to null every time, instead use a service to store the previous value of it(which can be null, or some solution) and assign that value when it gets reinitialized.

Reply:
Oh, I see -- that's interesting. An approach like that could be made to work but I think it's a bit fiddly and it also starts introducing multiple sources of truth, since we already store player info in PlayerTranscriptService.

How about doing something like the following: if player navigates to last card, look at PlayerTranscriptService. If the last thing there is a feedback thing that goes to a next state, show the help card immediately? Basically, use PlayerTranscriptService to store all the current status, including partial help cards and stuff.

Reply:
Thanks for the input. Will accommodate all information about help card in PlayerTranscriptService.